### PR TITLE
Print rand seed on failure, run tests in CI

### LINKED ISSUE #112: Tests: print seed when tests fail
In tests where randomization is used, the seed used in that test should be printed when the test fails. This would enable reproducibility for potentially flaky tests.

NOTE: Not directly dependent, but would be easier to implement after #108 is merged. We could then leverage [`checkpoint(string)`](https://stefantalpalaru.github.io/nim-unittest2/unittest2.html#checkpoint%2Cstring) from unittest2.

### PR DESCRIPTION
**Pull request by jromero**
_Built on top of https://github.com/MerosCrypto/Meros/pull/151 due to the changes to GH Actions._

This PR provides the following:
- Prints Randomize Seed when a failure occurs. (Resolves #112)
- Adds a GitHub Action which runs `nimble ci`.

As a bonus, it produces build artifacts that can be downloaded and executed on Linux.

### CODE REVIEW

File: Meros.nimble [hunk 1]
```diff
@@ -14,7 +14,7 @@ skipExt = @["nim"]
 
 #Dependencies
 requires "nim >= 1.0.4"
-requires "https://github.com/MerosCrypto/Argon2"
+requires "https://github.com/MerosCrypto/Argon2 >= 1.1.2"
```
Comment (kayabaNerve) [inline Meros.nimble]: Why was this needed? I know we have the same thing for ForceCheck, yet I honestly forget why.

File: Meros.nimble [hunk 2]
```diff
@@ -97,7 +97,7 @@ task unit, "Run unit tests.":
             .join(" ")
 
     #Ensure dependencies are installed.
-    nimbleExec "install --depsOnly"
+    nimbleExec "install --depsOnly -Y"
```
Comment (kayabaNerve) [inline Meros.nimble]: Since Nimble uses a single location for all Nimble packages, this can be dangerous. Can't we have the automated scripts run `nimble install --depsOnly -Y` in advance so this `nimble install --depsOnly` doesn't trigger a prompt?

File: tests/Fuzzed.nim [hunk 3]
```diff
@@ -1,29 +1,52 @@
...
+
+#Random standard lib.
+import random
+
 const TEST_FUZZING_LOW {.intdefine.}: int = 1
 const TEST_FUZZING_MID {.intdefine.}: int = 4
 const TEST_FUZZING_HIGH {.intdefine.}: int = 8
 
+
+template setupRandom =
```
Comment (kayabaNerve) [inline tests/Fuzzed.nim]: Please use empty parentheses.

File: tests/Fuzzed.nim [hunk 4]
```diff
@@ -1,29 +1,52 @@
...
+    randomize(seed)
+    checkpoint("Randomize Seed: " & $seed)
+
+template noFuzzTest*(
+    name: string,
+    body: untyped
+) =
+    test name:
+        setupRandom
+        body
```
Comment (kayabaNerve) [inline tests/Fuzzed.nim]: This isn't a comment for this PR, more the tests overall. While I like unittest more than what we had, I'm not a fan of it for a variety of reasons. I've been debating rewriting it, especially since your Nimble script handles test discovery (which is the only reason I want to move from the custom Python harness to Pytest). As a less extreme step, would it be optimal to write a custom `suite` macro to ensure no `test`s slip in?

File: tests/Fuzzed.nim [hunk 5]
```diff
@@ -1,29 +1,52 @@
 import unittest
```
Comment (kayabaNerve) [inline tests/Fuzzed.nim]: If every file now needs to import Fuzzed for noFuzzTest, we should export unittest so we don't we need to import both.

File: tests/Fuzzed.nim (hunk 3 above)
Comment (kayabaNerve) [reply tests/Fuzzed.nim]: Side note, very happy with how cleanly this was implemented. Good job :D I'd like to make sure it's noted this isn't thread safe, a concern you originally brought up. This is completely acceptable as we don't thread our tests as of right now (and can't, even with unittest2).

File: tests/Fuzzed.nim (hunk 5 above)
Comment (kayabaNerve) [reply tests/Fuzzed.nim]: To append to this comment, we should be able to explicitly export the suite macro, which also resolves a below comment about stopping accidental `test` usage.

File: .github/workflows/ci.yaml [hunk 6]
```diff
@@ -9,18 +9,53 @@ on:
...
+            -   name: Cache nimble
+                id: cache-nimble
+                uses: actions/cache@v1
+                with:
+                    path: ~/.nimble
+                    key: ${{ runner.os }}-nimble-stable
+            -   uses: jiro4989/setup-nim-action@v1.0.2
+                with:
+                    nim-version: '1.0.4'
+            -   name: Run unit tests
```
Comment (kayabaNerve) [inline .github/workflows/ci.yaml]: Incorrect name.

**Comment by kayabaNerve** [comment]
@jromero Any update to this?

File: Meros.nimble (hunk 1 above)
Comment (jromero) [reply Meros.nimble]: It's because `nimble` cache might have a previous incompatible version installed and would accurately satisfy the dependency. This was true here for `Argon2`. By setting a version, it then forces that version to be updated (automatically if I remember correctly) in the case that you have a previous version installed.

We should be doing this for all dependencies to be honest but didn't want to tie it in with the other changes here.

File: .github/workflows/ci.yaml (hunk 6 above)
Comment (jromero) [inline .github/workflows/ci.yaml]:
```suggestion
            -   name: Build Meros
```

File: Meros.nimble (hunk 1 above)
Comment (kayabaNerve) [reply Meros.nimble]: Thanks for the clarification. Since we're already doing it with Argon2, feel free to. Else, we can create a new thing immediately after this is merged.

File: tests/Fuzzed.nim (hunk 4 above)
Comment (jromero) [reply tests/Fuzzed.nim]: I'm by no means proficient in nim to be able to write a proper macro. I might give it a try in the future but not necessarily comfortable at this point in time. Additionally, I think other things might be more pressing.

**Comment by jromero** [comment]
This is ready for re-review. Good to merge?

File: tests/Fuzzed.nim (hunk 4 above)
Comment (kayabaNerve) [reply tests/Fuzzed.nim:25]: Only reason I bring it up is the Nim tests take forever due to not being threaded. Beyond that, unittest has a lot of quirks I'm not happy with. That said, yes, it's out of scope for this and something to discuss down the line. Thanks for keeping me in check.

**Comment by kayabaNerve** [comment]
LGTM. Thanks for doing this. Fixing https://github.com/MerosCrypto/Meros/issues/112 will make resolving test bugs much easier.

I did make one commit to remove a stray randomize call/not export the test macro to ensure it's not accidentally used, but everything else is great.

I'm running the tests one last time right now to ensure my commit doesn't break anything and then will merge.
